### PR TITLE
JS: Regex example

### PR DESCRIPTION
**Pull request by RunDevelopment**
This PR fixes a regex example which Prism does not highlight correctly because it is preceded by a comment.

### Before
image

### After
image

### CODE REVIEW

**Comment by mAAdhaTTah** [comment]
I will merge this fix, but should we file this as a bug for the JS language?

**Comment by RunDevelopment** [comment]
You're right that I just treated the symptoms of the underlying problem but that's because the underlying problem is not really fixable.

The regex in the _before_ state did not match because a regex literal has to be preceded by a character like `=`, `,`, `+`, `/`, etc. (__not__ including alphanumeric characters!) followed by any number of spaces (`\s`). But because of the single-line comment, the character preceding our regex is an `s`, so no match.

To correctly match the regex, we have to detect the preceding single-line comment.
_Just include a `//.*` into the lookbehind and done, right?_
Well, no. Consider the following example:

```js
var foo = 1, g = 2;
var a = /* // this will make our lives hard */ 5
/foo/g;
```
(GitHub gets this wrong. It thinks that `/foo/g` is a regex.)

`//` can also be part of strings and even regexes to make matters worse.

So to correctly match the regex we need to correctly match the preceding single-comment for which need to consider comments, strings, and regexes.

Put simply, it's impossible (IMO) to fix this bug with the methods Prism currently offers.
(And I don't see the point in opening an issue which we cannot fix.)

**Comment by mAAdhaTTah** [comment]
Fair enough. This also seems like something that won't come up a lot in practice. Thanks for explaining!
